src/components/ui/areaCard.tsx

```diff
@@ -15,11 +17,21 @@ const AreaCard = React.forwardRef<HTMLDivElement, AreaProps>((area, ref) => {
       title={area.name}
       className="flex h-full w-full flex-col items-center justify-center"
     >
-      <SheetTrigger>
-        <div className="absolute right-2 top-2 h-6 w-6 shrink-0 rounded-full hover:bg-accent hover:text-accent-foreground">
-          <Icons.info />
-        </div>
-      </SheetTrigger>
+      <div className="absolute right-2 top-2 space-x-1">
+        <Button
+          variant="icon"
+          size="icon"
+          className="h-4 w-4 opacity-50 shrink-0 rounded-full hover:bg-accent hover:text-accent-foreground"
+          onClick={area.toggleVisibility}
+        >
+          {area.visible ? <Icons.show /> : <Icons.hide />}
+        </Button>
+        <SheetTrigger>
+          <div className="h-4 w-4 opacity-50 shrink-0 rounded-full hover:bg-accent hover:text-accent-foreground">
+            <Icons.info />
+          </div>
+        </SheetTrigger>
+      </div>
```

Comment:
Joined all absolute elements in one absolute parent for consistent positioning / spacing